docs/adr/0006-process-for-onboarding-new-tenants.md

```diff
@@ -0,0 +1,299 @@
+ |           +-> (Operations storage configuration)
+ |           +-> (The Id of the OPERATIONS+Litware sub-tenant for the Operations service to use)
+ |
+ +-> Service Tenants
+       |
+       +-> WORKFLOW
+       |     |
+       |     +-> WORKFLOW+Litware
+       |           +-> (Operations storage configuration)
+       |           +-> (The Id of the OPERATIONS+WORKFLOW+Litware sub-tenant for the Operations service to use)
+       |           +-> (FooBar storage configuration)
+       |
+       +-> OPERATIONS
+       |     |
+       |     +-> OPERATIONS+WORKFLOW+Litware
+       |     |     +-> (FooBar storage configuration)
+       |     |
+       |     +-> OPERATIONS+Litware
+       |
+       +-> FOOBAR
+```
+
+Then, the Operations service will call out to the FooBar service to enroll the OPERATIONS+Litware service. As with enrolling the OPERATIONS+WORKFLOW+Litware service, this results in FooBar storage configuration being attached to the OPERATIONS+Litware service:
+
+```
+Root tenant
+ |
+ +-> Client Tenants
+ |     |
+ |     +-> Contoso
+ |     |
+ |     +-> Litware
+ |           +-> (Workflow storage configuration)
+ |           +-> (The Id of the WORKFLOW+Litware sub-tenant for the Workflow service to use)
+ |           +-> (Operations storage configuration)
+ |           +-> (The Id of the OPERATIONS+Litware sub-tenant for the Operations service to use)
+ |
+ +-> Service Tenants
+       |
+       +-> WORKFLOW
+       |     |
+       |     +-> WORKFLOW+Litware
+       |           +-> (Operations storage configuration)
+       |           +-> (The Id of the OPERATIONS+WORKFLOW+Litware sub-tenant for the Operations service to use)
+       |           +-> (FooBar storage configuration)
+       |
+       +-> OPERATIONS
+       |     |
+       |     +-> OPERATIONS+WORKFLOW+Litware
+       |     |     +-> (FooBar storage configuration)
+       |     |
+       |     +-> OPERATIONS+Litware
+       |           +-> (FooBar storage configuration)
+       |
+       +-> FOOBAR
+```
+
+This completes the enrollment of Litware to the Workflow and Operations services. As can be seen from the above, there are three different paths through which Litware makes indirect use of the FooBar service, and it's possible for the client to use separate storage for each. In fact, this will be the default; even if the client is using Marain storage, the data for their three different usage scenarios for FooBar will be stored in different containers.
+
+However, it's worth pointing out that the client does not get to configure these new sub-tenants directly. In fact, they will be unaware of them - they are essentially implementation details of our approach to multi-tenancy in Marain. They will not be able to retrieve the sub-tenants from the tenancy service or update them directly. That said, it's likely that the management API will allow the configuration to be changed - but without exposing the fact that these sub-tenants exist.
+
+### Offboarding
```

Comment:
Are we easily able to determine the entire dependency tree, so that if we are decommissioning all of the tenants, we don't have orphans, because we've lost track of them?

Reply:
Good question - we need to verify if DELETE of a child in its parent cascades that delete (which it should but I bet it doesn't) and when we get to this hypothetical offboarding thing, that it offboards all the children too.

You can walk that dependency tree, though.